File: python/grc_agent/mappings/framework_mapper.py
from __future__ import annotations

import os
from pathlib import Path
from typing import Any

from ..data.data_loader import load_json

_mappings_path = os.path.join(Path(__file__).parents[3], "data", "framework-mappings.json")


async def load_framework_mappings() -> dict[str, Any]:
    """Load the cross-framework control mappings file."""
    return await load_json(_mappings_path)

File: src/mappings/framework-mapper.ts
import path from "node:path";
import { loadJson } from "../data/data-loader.js";

type FrameworkMapping = {
  source: string;
  target: string;
  mappings: Array<{
    source_control_id: string;
    target_control_id: string;
  }>;
};

type FrameworkMappingsFile = {
  mappings: FrameworkMapping[];
};

const mappingsPath = path.resolve(process.cwd(), "data", "framework-mappings.json");

export async function loadFrameworkMappings(): Promise<FrameworkMappingsFile> {
  return loadJson<FrameworkMappingsFile>(mappingsPath);
}
